### PR TITLE
Fix #4499: Shorten internal module IDs in FewestModules mode

### LINKED ISSUE #4499: FewestModules can result in filenames that are greater than 255 Characters
So I have a private repository where I attempted to break a bunch of scalajs code into modules.  Alot of those modules reference  the same code, so it creates filenames longer than 255 characters which basically no filesystem supports.  I can't share the code but I do feel like someone on the maintainer team would understand how this might happen.  So I switched to smallest modules and the problem went away.

Happy to attempt to put something together but figured I'd file the issue first.

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: linker/shared/src/test/scala/org/scalajs/linker/MaxModuleSplittingTest.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  /** Smoke test to ensure modules do not get merged too much. */
+  @Test
+  def avoidsCollisions(): AsyncResult = await {
+    val classDefs = Seq(
+        mainTestClassDef({
+            consoleLog(str("Hello World!"))
+        })
+    )
```
Comment (sjrd) [inline linker/shared/src/test/scala/org/scalajs/linker/MaxModuleSplittingTest.scala:40]:
```suggestion
    val classDefs = Seq(
      mainTestClassDef({
        consoleLog(str("Hello World!"))
      })
    )
```

File: linker/shared/src/test/scala/org/scalajs/linker/MaxModuleSplittingTest.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    val expectedFiles = Set(
+        "internal-0.js", // public module
+        "internal-1.js", // public module
+        "internal-2.js"  // internal module, avoiding internal-0 and internal-1.
+    )
```
Comment (sjrd) [inline linker/shared/src/test/scala/org/scalajs/linker/MaxModuleSplittingTest.scala:46]:
```suggestion
    val expectedFiles = Set(
      "internal-0.js", // public module
      "internal-1.js", // public module
      "internal-2.js"  // internal module, avoiding internal-0 and internal-1.
    )
```

**Review by sjrd — CHANGES_REQUESTED** [review]
Only a few indentation issues. Otherwise LGTM.

**Comment by gzm0** [comment]
Updated.

**Review by sjrd — APPROVED** [review]

**Pull request merged**